pandas/tests/io/test_html.py

```diff
@@ -25,8 +25,23 @@
 import pandas.util._test_decorators as td
 from pandas.util.testing import makeCustomDataframe as mkdf, network
 
+HERE = os.path.dirname(__file__)
 
-DATA_PATH = tm.get_data_path()
+
+def pytest_generate_tests(metafunc):
```

Comment:
What do people thing about this approach?

This seems to be the recommended why to dynamically generate parametrized fixtures, based on some condition at runtime (the files present in this folder).

Reply:
Actually, in this case I'm not sure it makes sense...

When the files aren't present, `paths` will be empty...

Reply:
Actually, this probably is what we want. When the files are present, we'll get one fixture per file, which is exactly what we want. When they aren't present, the tests are skipped, which is again what we want.

Reply:
But will it then still fail in the strict case?

Reply:
The files won't be there, so no fixtures are generated. It won't "exist", so there's nothing to fail :)

Reply:
Yes, but we still *want* it to fail if there are no files and if `--strict-data-files` is passed?

Reply:
I think that's impossible for dynamically generated fixtures like this (which may be your point).

The alternative is to explicitly list them, which isn't so difficult, as there are only 8. Then we'll have a fixture like `html_files`. going forward, if we have to add an additional HTML file, we would add it there. I'd prefer being explicit in this case, since there are so few files.

Reply:
And the advantage of explicitly listing them is also that we don't need to use this dark pytest magic :-)

Reply:
Just the normal pytest magic :)

Reply:
Late here but I don’t mind this approach (not that different from metaclasses in Python). Is it not theoretically possible though to just fail the generated tests if the strict parameter is supplied yet no files are found?

Reply:
In this case, I'd prefer to explicitly list them since there are so few.